src/ipbb/cmds/repo.py

```diff
@@ -285,6 +285,14 @@ def git(env, repo, branch, dest):
         echo('Checking out branch/tag ' + style(branch, fg='blue'))
         sh.git('checkout', branch, '-q', _out=sys.stdout, _cwd=lRepoLocalPath)
 
+    elif revision is not None:
+        echo('Checking out revision ' + style(revision, fg='blue'))
```

Comment:
The revision existence can and should be checked beforehand, similarly to what's done for branches/tags.
`git ls-remote` is your friend here, with syntax 
```sh
git ls-remote <repo> <hash>
```
or
```python
sh.git('ls-remote', repo, branch)
```

Reply:
Unfortunately not. ```ls-remote``` only works on references and not on revisions.

Reply:
Ah, right you are. I was too hasty consulting the `ls-remote` doc.